weave-js/src/react.tsx

```diff
@@ -553,15 +553,20 @@ const RE_WEAVE_CALL_REF_PATHNAME = new RegExp(
 
 export const parseRef = (ref: string): ObjectRef => {
   const url = new URL(ref);
-  let splitLimit: number;
+  let maxSplitsToMake: number;
+  let allowableNumSeparations: number[];
 
   const isWandbArtifact = url.protocol.startsWith('wandb-artifact');
   const isLocalArtifact = url.protocol.startsWith('local-artifact');
   const isWeaveRef = url.protocol.startsWith('weave');
   if (isWandbArtifact) {
-    splitLimit = 4;
+    maxSplitsToMake = 4;
+    // artifact URI expected as <entity>/<project>/<artifact name>:<artifact version - ie, v0>/<optional path>
```

Comment:
expects in this form where version is v0, v1, or an artifact alias because that is how the SDK currently supports downloading an artifact (see https://docs.wandb.ai/guides/artifacts/download-and-use-an-artifact/). if we want to be able to reference the specific artifact version by digest on both the models and weave side, I think we'd need to add SDK support for retrieving an artifact by digest